intdata <- read.table(file = "/media/emese/331109B65C43AFD8/projects_2017/cbp_se_2017/rna-seq/intronnum/Arabidopsis_005_log2_1updown_intronnumber_nofilter.txt", header = F)
getPalette = colorRampPalette(brewer.pal(4, "GnBu"))
intdata$V1 <- factor(intdata$V1, levels=c("Arabidopsis_genomic_all", "Arabidopsis_se1_total_all", "Arabidopsis_se1_total_down", "Arabidopsis_se1_total_up"))
ggplot(data = intdata, aes(x = V1, y = V2, fill = intdata$V1)) + geom_boxplot() + ggtitle(label = "Intron numbers of Arabidopsis thaliana WT and se1 total RNA seq libraries") + theme_classic() + theme(plot.title=element_text(size = 20)) + xlab(label = "") + ylab(label = "Number of introns") + scale_x_discrete(labels = c("All sig.\n unfiltered genes", "Downregulated", "Upregulated")) + scale_fill_manual(values = rev(getPalette(4)))
ggplot(data = intdata, aes(x = V1, y = V2, fill = intdata$V1)) + geom_boxplot(outlier.colour = NA) + ggtitle(label = "Intron numbers of Arabidopsis thaliana\nWT and se1 total RNA seq libraries(padj<0.05)") + theme_classic() + theme(plot.title=element_text(size = 20)) + xlab(label = "") + ylab(label = "Number of introns") + scale_x_discrete(labels = c("Genomic background", "All sig.\n unfiltered genes", "Downregulated", "Upregulated")) + scale_fill_manual(values = rev(getPalette(4))) + geom_signif(comparisons = list(c("Arabidopsis_genomic_all", "Arabidopsis_se1_total_all"), c("Arabidopsis_genomic_all", "Arabidopsis_se1_total_down"), c("Arabidopsis_genomic_all", "Arabidopsis_se1_total_up")), map_signif_level =TRUE, step_increase= 0.1, y_position =c(19, 23, 27),test = "wilcox.test") + coord_cartesian(ylim=c(0, 30))
intdata_up <- read.table(file = "../cbp_se_2017/rna-seq/intronnum/deseq2_WT_se1_intronnum_005_log2_1up_nofiler.txt", header = F)
intdata_down <- read.table(file = "../cbp_se_2017/rna-seq/intronnum/deseq2_WT_se1_intronnum_005_log2_1down_nofiler.txt", header = F)
intdata_bg <- read.table(file = "../cbp_se_2017/rna-seq/intronnum/arabidopsis_thaliana_iso1_intronnum.txt", header = F)
wilcox.test(intdata_bg$V2, intdata_up$V3, "two.sided")
wilcox.test(intdata_bg$V2, intdata_up$V3, "less")
wilcox.test(intdata_bg$V2, intdata_up$V3, "greater")
wilcox.test(intdata_bg$V2, intdata_down$V3, "two.sided")
wilcox.test(intdata_bg$V2, intdata_down$V3, "less")
wilcox.test(intdata_bg$V2, intdata_down$V3, "greater")

getPalette = colorRampPalette(brewer.pal(4, "GnBu"))
intdata$V1 <- factor(intdata$V1, levels=c("Arabidopsis_genomic_all", "Arabidopsis_se1_total_all", "Arabidopsis_se1_total_down", "Arabidopsis_se1_total_up"))
ggplot(data = intdata, aes(x = V1, y = V2, fill = intdata$V1)) + geom_boxplot() + ggtitle(label = "Intron numbers of Arabidopsis thaliana WT and se1 total RNA seq libraries") + theme_classic() + theme(plot.title=element_text(size = 20)) + xlab(label = "") + ylab(label = "Number of introns") + scale_x_discrete(labels = c("All sig.\n unfiltered genes", "Downregulated\n (log2FC>=1)", "Upregulated\n (log2FC<=-1)")) + scale_fill_manual(values = rev(getPalette(4)))
ggplot(data = intdata, aes(x = V1, y = V2, fill = intdata$V1)) + geom_boxplot(outlier.colour = NA) + ggtitle(label = "Intron numbers of Arabidopsis thaliana\nWT and se1 total RNA seq libraries(padj<0.05)") + theme_classic() + theme(plot.title=element_text(size = 20)) + xlab(label = "") + ylab(label = "Number of introns") + scale_x_discrete(labels = c("Genomic background", "All sig.\n unfiltered genes", "Downregulated\n (log2FC>=1)", "Upregulated\n (log2FC<=-1)")) + scale_fill_manual(values = rev(getPalette(4))) + geom_signif(comparisons = list(c("Arabidopsis_genomic_all", "Arabidopsis_se1_total_all"), c("Arabidopsis_genomic_all", "Arabidopsis_se1_total_down"), c("Arabidopsis_genomic_all", "Arabidopsis_se1_total_up")), map_signif_level =TRUE, step_increase= 0.1, y_position =c(19, 23, 27),test = "wilcox.test") + coord_cartesian(ylim=c(0, 30))
intdata_up <- read.table(file = "../cbp_se_2017/rna-seq/intronnum/deseq2_WT_se1_intronnum_005_log2_1up.txt", header = F)
intdata_down <- read.table(file = "../cbp_se_2017/rna-seq/intronnum/deseq2_WT_se1_intronnum_005_log2_1down.txt", header = F)
intdata_bg <- read.table(file = "../cbp_se_2017/rna-seq/intronnum/arabidopsis_thaliana_iso1_intronnum.txt", header = F)
wilcox.test(intdata_bg$V2, intdata_up$V3, "two.sided")
wilcox.test(intdata_bg$V2, intdata_up$V3, "less")
wilcox.test(intdata_bg$V2, intdata_up$V3, "greater")
wilcox.test(intdata_bg$V2, intdata_down$V3, "two.sided")
wilcox.test(intdata_bg$V2, intdata_down$V3, "less")
wilcox.test(intdata_bg$V2, intdata_down$V3, "greater")
